### DISCUSSION TITLE
problem vacuum xiaomi

### DISCUSSION THREAD

**Discussion by enzob1973** [rytilahti/python-miio#928] [General]
hi i have a problem with xiaomi vacuum model MJSTG1 MI ROBOT VACUUM-MOP ESSENTIAL. I tried:
vacuum:
 - platform: xiaomi_miio
   host: 10.0.0.134
   token: my token
   name: 'Living Vacuum'

Home assistant doesn't see him. Help

**Comment by rytilahti** [comment]
Hi, I think that this vacuum cleaner is not from rockrobo, which is the only manufacturer supported by the homeassistant integration at the moment. There are some custom components available on the internet, but I'm not sure if your device is supported by one of those, sorry!

**Reply by radek-kafka** [reply]
I would try g1vacuum as the commend.. this one is working with mine (detected as mijia.vacuum.v2)

**Comment by awalasz** [comment]
Hi, @rytilahti  I was searching for discussion, but I can't find any information. 
Are there plans to support other vacuums in the home assistant integration? Is there any issue I could subscribe to for future updates on that?

**Reply by rytilahti** [reply]
Hi @awalasz yes, the plan is to introduce support for all devices supported by this library as well as for any devices using the miot protocol for local controls. #1114 and #1495 are tangentially related, and so is #1581 (support for miot devices) but there is currently no meta issue to track the progress for the homeassistant integration as it's mostly done simultaneously those changes listed in those two issue.

**Reply by awalasz** [reply]
Thanks for quick reply! :)
